libs/utils/utils.js

```diff
@@ -1038,14 +990,26 @@ async function checkForPageMods() {
     mepButton,
     martech,
   } = Object.fromEntries(PAGE_URL.searchParams);
+  let targetInteractionData = null;
   if (mepParam === 'off') return;
   const pzn = getMepEnablement('personalization');
   const promo = getMepEnablement('manifestnames', PROMO_PARAM);
   const target = martech === 'off' ? false : getMepEnablement('target');
   const xlg = martech === 'off' ? false : getMepEnablement('xlg');
+  const enablePersV2 = enablePersonalizationV2();
+
   if (!(pzn || target || promo || mepParam
     || mepHighlight || mepButton || mepParam === '' || xlg)) return;
-  if (target || xlg) {
+
+  if (martech !== 'off' && (target || xlg || pzn) && enablePersV2) {
+    const { locale } = getConfig();
+    try{
+      targetInteractionData = await loadAnalyticsAndInteractionData({ locale });
```

Comment:
We don't need to `await` this promise here. We should fire it here, store the promise and pass it to `personalization.js`. That way we won't wait until this data is loaded to start loading `personalization.js`. they both can be fired almost in parallel.  Then we can await the promise right before we need it inside `personalization.js`

Reply:
Granted you'd have to also move the error handling logic if you did that.